Signal/src/ViewControllers/SelectRecipientViewController.m

```diff
@@ -0,0 +1,553 @@
+{
+
+    [self updateCountryWithName:countryName callingCode:callingCode countryCode:countryCode];
+
+    [self textField:self.phoneNumberTextField shouldChangeCharactersInRange:NSMakeRange(0, 0) replacementString:@""];
+}
+
+#pragma mark - UITextFieldDelegate
+
+// TODO: This logic resides in both RegistrationViewController and here.
+//       We should refactor it out into a utility function.
+- (BOOL)textField:(UITextField *)textField
+    shouldChangeCharactersInRange:(NSRange)range
+                replacementString:(NSString *)insertionText
+{
+    [ViewControllerUtils phoneNumberTextField:textField
+                shouldChangeCharactersInRange:range
+                            replacementString:insertionText
+                                  countryCode:_callingCode];
+
+    [self updatePhoneNumberButtonEnabling];
+
+    return NO; // inform our caller that we took care of performing the change
+}
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField
+{
+    [textField resignFirstResponder];
+    [self tryToSelectPhoneNumber];
+    return NO;
+}
+
+#pragma mark - Table Contents
+
+- (void)updateTableContents
+{
+    OWSTableContents *contents = [OWSTableContents new];
+
+    __weak SelectRecipientViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
+
+    OWSTableSection *phoneNumberSection = [OWSTableSection new];
+    phoneNumberSection.headerTitle = [self.delegate phoneNumberSectionTitle];
+    const CGFloat kCountryRowHeight = 50;
+    const CGFloat kPhoneNumberRowHeight = 50;
+    const CGFloat kButtonRowHeight = 60;
+    [phoneNumberSection addItem:[OWSTableItem itemWithCustomCellBlock:^{
+        SelectRecipientViewController *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return (UITableViewCell *)nil;
+        }
+
+        UITableViewCell *cell = [UITableViewCell new];
+
+        // Country Row
+        UIView *countryRow = [self createRowWithHeight:kCountryRowHeight previousRow:nil superview:cell.contentView];
+        [countryRow
+            addGestureRecognizer:[[OWSAnyTouchGestureRecognizer alloc] initWithTarget:self
+                                                                               action:@selector(countryRowTouched:)]];
+
+        UILabel *countryCodeLabel = self.countryCodeLabel;
+        [countryRow addSubview:countryCodeLabel];
+        [countryCodeLabel autoPinEdgeToSuperviewEdge:ALEdgeLeft withInset:20.f];
+        [countryCodeLabel autoVCenterInSuperview];
+
+        [countryRow addSubview:self.countryCodeButton];
+        [self.countryCodeButton autoPinEdgeToSuperviewEdge:ALEdgeRight withInset:20.f];
+        [self.countryCodeButton autoVCenterInSuperview];
+
+        // Phone Number Row
+        UIView *phoneNumberRow =
+            [self createRowWithHeight:kPhoneNumberRowHeight previousRow:countryRow superview:cell.contentView];
+        [phoneNumberRow addGestureRecognizer:[[OWSAnyTouchGestureRecognizer alloc]
```

Comment:
I don't think we should use "AnyTouch" for "selecting" items (as in from a list) or "pushing buttons". Users know to tap in this case (with it's tiny delay).

One small reason to use a tap is that it comes with a built in "undo" (e.g. slide away from the the thing you accidentally tapped on before lifting your finger). I don't think many people do this consciously, but whenever you go against the grain of something as fundamentally implemented as a standard tap gesture you run the risk of unexpected consequences.

I do like how we're using it to hide the call controls, but not so much because it's fast, rather it's a place where people actually *might* be swiping to try to "dismiss" the controls.